### PR TITLE
Clarify link font-weight in table documenation

### PR DESCRIPTION
**Pull request by heatherlarsen**
### :pushpin: Summary

If merged, this PR provides clarification around style recommendations for links within tables. The existing guidance is leading to some confusion because the language used in the documentation is specific to Figma styles, e.g., `Body/200/Link`, that are not available in code ([context](https://hashicorp.slack.com/archives/C7KTUHNUS/p1704745450195049)).

### :camera_flash: Screenshots

**Before:**
Screenshot 2024-01-08 at 1 20 57 PM

**After:**
Screenshot 2024-01-08 at 1 18 32 PM

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A11y tests have been run locally (`yarn test:a11y --filter="COMPONENT-NAME"`)
- [ ] If documenting a new component, an acceptance test that includes the `a11yAudit` has been added
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/docs/components/table/partials/content/link-usage.md [hunk 1]
```diff
@@ -1,24 +1,19 @@
...
-Use `Body / 200 / Link` as the default typographic style within a table. This style increases the prominence slightly to differentiate it from other text content.
+To increases the prominence and further differentiate links from other text content, we recommend using a font-weight of `medium`. In code, you can use the CSS helper classes `hds-typography-body-200` and `hds-font-weight-medium`. In Figma, you can use the text style `Body/200/Link`.
 
 ![Link example](/assets/components/table/link-example.png)
 
-#### Multiple links
+### Multiple links
 
-If a table contains more than one column of links, consider using `Body / 200 / Link` for the most important links; usually the title of the row, ID, or other naming convention. For less important links, use `Body / 200 / Regular` with an added underline in Figma.
+If a table contains more than one column of links, consider using a font-weight of `medium` for the most important links; usually the title of the row or ID. For less important links, use a font-weight of `regular`.
```
Comment (alex-ju) [inline website/docs/components/table/partials/content/link-usage.md:11]:
```suggestion
If a table contains more than one column of links, consider using a font-weight of `medium` for the most important links; usually the title of the row or ID. For less important links, use a `font-weight` of `regular`.
```

File: website/docs/components/table/partials/content/link-usage.md [hunk 2]
```diff
@@ -1,24 +1,19 @@
...
-!!! Info
+By default, we recommend using `secondary` [Inline Links](/components/link/inline) within a table, to avoid overloading the UI with the actionable blue color. The `secondary` variant uses the `foreground-strong` color.
 
-This guidance is an extension of the [Inline Link](/components/link/inline) guidelines.
-!!!
-
-Within a table, use `secondary` (`Foreground / Strong`) links as the default.
-
-Use `Body / 200 / Link` as the default typographic style within a table. This style increases the prominence slightly to differentiate it from other text content.
+To increases the prominence and further differentiate links from other text content, we recommend using a font-weight of `medium`. In code, you can use the CSS helper classes `hds-typography-body-200` and `hds-font-weight-medium`. In Figma, you can use the text style `Body/200/Link`.
```
Comment (alex-ju) [inline website/docs/components/table/partials/content/link-usage.md:5]:
```suggestion
To increase the prominence and further differentiate links from other text content, we recommend using a font-weight of `medium`. In code, you can use the CSS helper classes `hds-typography-body-200` and `hds-font-weight-medium`. In Figma, you can use the text style `Body/200/Link`.
```

**Review by alex-ju — APPROVED** [review]
Looks good. Left a few suggestions before getting it in.

**Review by MelSumner — APPROVED** [review]

**Comment by MelSumner** [comment]
Noting that this is not what we have in the HDS showcase right now, so that probably should be updated (even if just for our own reference). I personally didn't realize this was our guidance.

**Review by KristinLBradley — APPROVED** [review]

**Pull request merged**